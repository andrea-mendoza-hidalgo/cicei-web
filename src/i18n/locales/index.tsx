import en from './translations.en.json';
import es from './translations.es.json';
import { SharedEn, SharedEs } from '../../shared/components/locale';
import { PagesEn, PagesEs } from '../../components/pages/locales';
import ProjectEn from '../../components/project/locale/project.en.json';
import ProjectEs from '../../components/project/locale/project.es.json';
import PublicationsEn from '../../components/publication/locale/publication.en.json';
import PublicationsEs from '../../components/publication/locale/publication.es.json';
import ResearchersEn from '../../components/researchers/locale/researchers.en.json';
import ResearchersEs from '../../components/researchers/locale/researchers.es.json';
import NewsEs from '../../components/news/locale/news.es.json';
import NewsEn from '../../components/news/locale/news.en.json';

en.shared= SharedEn;
es.shared= SharedEs;
en.pages = PagesEn;
es.pages = PagesEs;
en.project = ProjectEn;
es.project = ProjectEs;
en.publication = PublicationsEn;
es.publication = PublicationsEs;
en.researchers = ResearchersEn;
es.researchers = ResearchersEs;
en.news = NewsEn;
es.news = NewsEs;



export {en, es};