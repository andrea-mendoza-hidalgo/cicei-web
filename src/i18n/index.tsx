import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import LanguageDetector from 'i18next-browser-languagedetector';
import { es, en } from './locales';

const options = {
  debug: false,

  resources: {
    'es-ES': {
      shared: es.shared,
      pages: es.pages,
      project: es.project,
      publication: es.publication,
      researchers: es.researchers,
      news: es.news
    },
    'en-US': {
      shared: en.shared,
      pages: en.pages,
      project: en.project,
      publication: en.publication,
      researchers: en.researchers,
      news: en.news
    }
  },

  fallbackLng: 'es-ES',

  ns: ['shared', 'pages', 'project', 'publication'],

  defaultNS: 'shared',

  react: {
    wait: true
  }
};

i18n.use(initReactI18next).use(LanguageDetector).init(options);

export default i18n;
