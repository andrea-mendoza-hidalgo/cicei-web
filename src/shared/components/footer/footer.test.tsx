import React from 'react';
import { mount } from 'enzyme';
import Footer from './footer';
import '../../../i18n';

describe('Footer Render Test', () => {
  it('Footer renders correctly ', () => {
    let footer = mount(<Footer/>)
    expect(footer).toMatchSnapshot();
  });
});
