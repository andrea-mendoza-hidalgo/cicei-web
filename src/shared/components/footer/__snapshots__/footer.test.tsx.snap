// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Render Test Footer renders correctly  1`] = `ReactWrapper {}`;
